### PR TITLE
Allow timeseries to be sliced by specific days

### LINKED ISSUE #629: Make graphs' x-axis rolling 30 days instead of starting from 30 Jan
Considering that the number of cases in India was fairly dormant in the first month, showing data from the beginning doesn't make much sense. This is especially true for the linear scale now that number of cases is pretty high.

I believe a 30 day (or some other number) rolling representation of data will be cleaner and easier to see. Can probably add that as an option.

### PR DESCRIPTION
**Pull request by umanghome**
**Description of PR**
Allows the Spread Trends timeseries to be shown as the last 7 days, last 30 days, and since forever.

**Type of PR**

- [ ] Bugfix
- [x] New feature

**Relevant Issues**  
Fixes #629 

**Checklist**

- [x] Compiles and passes lint tests
- [x] Properly formatted
- [x] Tested on desktop
- [x] Tested on phone

**Screenshots**

Desktop

Mobile: Google Chrome on Android

### CODE REVIEW

File: src/components/timeseries.js [hunk 1]
```diff
@@ -495,6 +503,32 @@ function TimeSeries(props) {
           />
         </div>
       </div>
+
+      <div className="pills" style={{marginTop: '32px', textAlign: 'right'}}>
```
Comment (umanghome) [inline src/components/timeseries.js:507]: The SVG above these controls overflows and pointer events are intercepted, leading to these buttons not receiving click-events. Gave `margin-top` to work around this issue.

**Comment by shuklaayush** [comment]
@umanghome I think you need to adjust the number of ticks on the x-axis. Right now, the ticks overlap when the 1-month tab is selected. It should be an easy fix in d3. Apart from that, looks good!

image

**Comment by umanghome** [comment]
@shuklaayush - I made some changes in [this commit](https://github.com/covid19india/covid19india-react/pull/712/commits/cfa223e077077f64f0ff7c5c3f7d0adce2e0bb78), but I'm not sure if that's the correct way to do this.

**Comment by jeremyphilemon** [comment]
@umanghome thank you for the PR :)

**Comment by umanghome** [comment]
@jeremyphilemon When would this be merged to master?

**Comment by shuklaayush** [comment]
@umanghome I'm handling this. It will be merged after #733. I'll fix the merge conflicts that arise.

**Comment by umanghome** [comment]
@shuklaayush Alright

**Pull request merged**